### PR TITLE
Build docker image periodically

### LINKED ISSUE #31: Pinning to specific phpstan versions with image tags does not work
see https://github.com/async-aws/aws/runs/749139572?check_suite_focus=true#step:5:1
It installs version 0.12.23 of your docker image, but still gets 0.12.26 of phpstan inside it.

This is because your Dockerfile uses a semver range (`^0.12.23` at the time of the 0.12.23 tag) to install phpstan. And so the selected version might be newer if the build runs after a newer version is released.

### PR DESCRIPTION
**Pull request by Nyholm**
This will fix #31 

This will run every 6 hours. It will tag add push a docker image. This needs us generating an application password. 

One can of course expand this later to automatically tag the repo too.

### CODE REVIEW

**Comment by OskarStark** [comment]
@Nyholm whenever I update the PHPStan version in the `Dockerfile`, I tag a repo release afterwards, which results in a tagged docker image.

I don't get what you want to achieve 🤔 

https://hub.docker.com/r/oskarstark/phpstan-ga/tags

If the goal is to have a docker tag per version, this should already be done, or am I missing sth. ?

**Comment by Nyholm** [comment]
Somehow there is something wrong with this process. See #31

File: .github/workflows/watch.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+          last=$(curl -s https://packagist.org/packages/phpstan/phpstan.json|jq '[.package.versions[]|select(.version|test("^\\d+\\.\\d+\\.\\d+$"))|.version]|max_by(.|[splits("[.]")]|map(tonumber))')
+          last=$(echo $last | tr -d '"')
+          echo "Last PHPStan version is $last"
+          echo ::set-output name=last::$last
+
+          sed -i -re "s/require phpstan\/phpstan [^\\]+/require phpstan\/phpstan $last /" Dockerfile
+          cat Dockerfile
+
+      - name: Docker login
+        run: echo '${{ secrets.DOCKER_PASSWORD }}' | docker login --username ${{ secrets.DOCKER_USERNAME }} --password-stdin
```
Comment (localheinz) [inline .github/workflows/watch.yml:30]: @OskarStark 

Are these secrets defined?

**Review by localheinz — APPROVED** [review]
👍

File: .github/workflows/watch.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+name: Watch
+
+on:
+  schedule:
+    - cron: '0 2,8,14,20 * * *'
```
Comment (localheinz) [inline .github/workflows/watch.yml:5]: We could miss versions here when multiple versions are tagged within 6 hours, but it's probably not a big problem.

File: .github/workflows/watch.yml (hunk 1 above)
Comment (OskarStark) [reply .github/workflows/watch.yml:30]: not yet

**Comment by Nyholm** [comment]
Psalm has now added a script similar to this PR. 

https://github.com/psalm/psalm-github-actions/runs/1164066545?check_suite_focus=true

**Comment by OskarStark** [comment]
Superseded by #34

**Pull request closed without merging**